### PR TITLE
feat: update to Manifest V3

### PR DESCRIPTION
**Pull request by jzempel**
- [ ] **BREAKING CHANGE?** 

## Description

Following notification from Google and in prep for [MV2 sunset](https://developer.chrome.com/docs/extensions/mv3/mv2-sunset/), update Garden Inspect to [Manifest V3](https://developer.chrome.com/docs/extensions/mv3/intro/).

## Detail

To be published as a 1.1.0 feature release to the Chrome web store.

### CODE REVIEW

File: src/scripts/on.ts [hunk 1]
```diff
@@ -92,24 +99,28 @@ declare const GARDEN_VERSION: string;
...
-      inspect(iframe.contentDocument, components);
-    } else {
-      console.log('Garden Inspect is unable to access cross-origin iframe:', iframe);
+    for (const iframe of iframes) {
+      if (iframe.contentDocument) {
+        inspect(iframe.contentDocument, components, request.latestVersion);
+      } else {
+        console.log('Garden Inspect is unable to access cross-origin iframe:', iframe);
+      }
     }
```
Comment (Francois-Esquire) [inline src/scripts/on.ts:115]: I think in the future, we can simplify the extension to be only the service worker:

```
chrome.scripting.executeScript(
    {
      target: {tabId: tabId, allFrames: true},
      func: isOn? off : on ,
    }
);
```
and remove the for loop for the accessing the frames.

File: src/background.ts [hunk 2]
```diff
@@ -50,7 +62,7 @@ const gardenInspect = (tabId?: number, toggle?: boolean): void => {
   }
```
Comment (Francois-Esquire) [inline src/background.ts:62]: I think the function signature for [`chrome.storage.local.get` requires passing in `null` as the first parameter.](https://developer.chrome.com/docs/extensions/reference/storage/#method-StorageArea-get)

File: src/scripts/on.ts (hunk 1 above)
Comment (jzempel) [reply src/scripts/on.ts:115]: Cool. Let's add this to the backlog @Francois-Esquire

File: src/background.ts [hunk 3]
```diff
@@ -20,21 +20,33 @@ const gardenInspect = (tabId?: number, toggle?: boolean): void => {
...
+    const targetTabId = tabId === undefined ? chrome.tabs.TAB_ID_NONE : tabId;
+
     chrome.storage.local.set({ [key]: on }, () => {
       if (on) {
         version().then(
           value => {
-            chrome.browserAction.setIcon({ path: 'images/on.png', tabId });
-            void chrome.tabs.executeScript({ code: `window.GARDEN_VERSION = '${value}';` });
-            void chrome.tabs.executeScript({ file: 'scripts/on.js' });
+            chrome.action.setIcon({ path: 'images/on.png', tabId });
```
[the remark this answers is not in the document]
Comment (jzempel) [reply src/background.ts:29]: Unlike the `executeScript` `target`, the `setIcon` function accepts an `undefined` `tabId`, so prefer to leave as-is.

File: src/background.ts (hunk 2 above)
Comment (jzempel) [reply src/background.ts:62]: The `get` function provides [multiple signatures](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/86fbf93e9dc4282914d59abdde8a0503543be1df/types/chrome/index.d.ts#L7420-L7441). The current use is valid (and the extension works as expected with `yarn start`).

File: src/scripts/on.ts (hunk 1 above)
Comment (Francois-Esquire) [reply src/scripts/on.ts:115]: Will do!

File: src/background.ts (hunk 2 above)
Comment (Francois-Esquire) [reply src/background.ts:62]: Ahhh good find.

**Review by Francois-Esquire — APPROVED** [review]

**Pull request merged**